### PR TITLE
python312Packages.pandoc-latex-environment: init at 1.1.6.4

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

Fix:
- #326592

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/nemosis/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    joblib
+    openpyxl
+    pandas
+    pyarrow
+    requests
+    xlrd
+  ];
+
+  pythonImportsCheck = [ "nemosis" ];
+  dontCheck = true; # require network and patching
```
Comment (natsukium) [inline pkgs/development/python-modules/nemosis/default.nix:41]:
```suggestion
  doCheck = false; # require network and patching
```

File: pkgs/development/python-modules/pandoc-latex-environment/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  src = fetchFromGitHub {
+    owner = "chdemko";
+    repo = "pandoc-latex-environment";
+    rev = "refs/tags/${version}";
+    hash = "sha256-DLuLFCKP5ip3duHlGAZnFip7GCDzWK91bka3DrknQSA=";
+  };
+
+  build-system = [ poetry-core poetry-dynamic-versioning ];
+  dependencies = [ panflute nemosis ];
```
Comment (natsukium) [inline pkgs/development/python-modules/pandoc-latex-environment/default.nix:25]:
```suggestion
  dependencies = [ panflute ];
```
unused
https://github.com/chdemko/pandoc-latex-environment/blob/a119d4c630460de5685bc17913ea5b07c6de0e11/pyproject.toml#L41-L43

File: pkgs/development/python-modules/feather-format/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  build-system = [ setuptools ];
+  dependencies = [ pyarrow ];
+
+  pythonImportsCheck = [ "feather" ];
+
+  meta = {
+    description = "Simple wrapper library to the Apache Arrow-based Feather File Format";
+    homepage = "https://github.com/wesm/feather";
+    license = lib.licenses.apsl20;
```
Comment (natsukium) [inline pkgs/development/python-modules/feather-format/default.nix:28]:
```suggestion
    license = lib.licenses.asl20;
```
`apsl20` is deprecated.
https://github.com/NixOS/nixpkgs/blob/f9d80213984daaf1a375f4199b4f666ce5bd21d3/lib/licenses.nix#L1321-L1326

File: pkgs/development/python-modules/feather-format/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    inherit pname version;
+
+    hash = "sha256-RfZ+N0XTlNTxYMptY2u/1Pi2jQEZncFkm25IfT6HiQM=";
+  };
+
+  build-system = [ setuptools ];
+  dependencies = [ pyarrow ];
+
+  pythonImportsCheck = [ "feather" ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/feather-format/default.nix:24]: Please disable tests explicitly.

File: pkgs/development/python-modules/nemosis/default.nix [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+  meta = {
+    description = "Downloader of historical data published by the Australian Energy Market Operator";
+    homepage = "https://github.com/UNSW-CEEM/NEMOSIS";
+    license = lib.licenses.gpl3Plus;
```
Comment (natsukium) [inline pkgs/development/python-modules/nemosis/default.nix:46]:
```suggestion
    license = lib.licenses.gpl3Only;
```
There is no reference to "Plus" in the source code.

File: pkgs/development/python-modules/panflute/default.nix [hunk 6]
```diff
@@ -26,6 +27,11 @@ buildPythonPackage rec {
 
   pythonImportsCheck = [ "panflute" ];
 
+  postPatch = ''
+    substituteInPlace panflute/tools.py \
+      --replace-fail "PANDOC_PATH = None" "PANDOC_PATH = '${pandoc}/bin/pandoc'"
+  '';
+
```
Comment (natsukium) [inline pkgs/development/python-modules/panflute/default.nix:34]: Why do you want to provide pandoc at build time?

**Comment by natsukium** [comment]
Finally, please reformat with `nixfmt-rfc-style`.

File: pkgs/development/python-modules/panflute/default.nix (hunk 6 above)
Comment (Sigmanificient) [reply pkgs/development/python-modules/panflute/default.nix:34]: should i use 
```py
  preCheck = ''
    export PATH="$PATH:${pandoc}/bin"
  '';
```
in `pandoc-latex-environment` instead?

File: pkgs/development/python-modules/pandoc-latex-environment/default.nix (hunk 2 above)
Comment (Sigmanificient) [reply pkgs/development/python-modules/pandoc-latex-environment/default.nix]: mistake, moved to #326686

File: pkgs/development/python-modules/panflute/default.nix (hunk 6 above)
Comment (natsukium) [reply pkgs/development/python-modules/panflute/default.nix:34]: Just add `pandoc` to `nativeCheckInputs`.

File: pkgs/development/python-modules/pandoc-latex-environment/default.nix [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+  pythonImportsCheck = [ "pandoc_latex_environment" ];
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  preCheck = ''
+    export PATH="$PATH:${pandoc}/bin"
+  '';
+
+  meta = {
+    description = "Pandoc filter for adding LaTeX environement on specific div";
```
Comment (natsukium) [inline pkgs/development/python-modules/pandoc-latex-environment/default.nix:38]:
```suggestion
    description = "Pandoc filter for adding LaTeX environment on specific div";
```

**Pull request merged**